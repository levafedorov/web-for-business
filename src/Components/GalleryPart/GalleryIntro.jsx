import React from 'react';



export default function GalleryIntro(props){


   return <div className="gallery-intro">
       <h2>Our gallery</h2>
       <p>Lorem, ipsum dolor sit amet consectetur adipisicing elit.
            Doloremque culpa facilis dicta enim ipsum fugit reiciendis doloribus 
            mollitia incidunt quasi. Lorem ipsum dolor sit amet consectetur adipisicing elit.
             Facilis reprehenderit cupiditate labore
             nobis neque ipsam, praesentium ratione vero sapiente laborum?
      </p>
   </div>
}