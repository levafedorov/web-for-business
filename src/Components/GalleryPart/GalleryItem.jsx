import React from 'react';



export default function GalleryItem(props){

   const {photo, title, width, active} = props;

    
   const bgImageStyle = {
       width: `${width}%`,
       backgroundImage: `url("${photo}")`
   }

   const titleUp = {
       top:0
   }

   return <div className="gallery-item" style={bgImageStyle}>
       <div className="title" style={active ? titleUp : null}>
           <h3>{title}</h3>
       </div>
   </div>
}


GalleryItem.defaultProps = {
     photo: '',
     title: 'No photo',
     width: "0",
     active: true
}
