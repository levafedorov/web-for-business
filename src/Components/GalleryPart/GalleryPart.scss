$white: white;
$black: black;
$dark-gray: #898d8e;
$light-gary: #b7bcbe;
$orange: #d7794a;
$whiten-gray:#e8e6e6;
$swampy: #1f3138;


@mixin decorator($bg:none, $bord:none){
   background-color: $bg;
   border: $bord;
}


@mixin bg-image($bg){
  background-image: url($bg);
  background-repeat: no-repeat;
  background-size: cover;
}



@mixin flex($just:center, $al:center){
   display: flex;
     justify-content: $just; 
     align-items: $al;
  }

  @mixin flex-self($just:center, $al:center){
     justify-self: $just; 
     align-self: $al;
  }


  @mixin myriad($size:1em, $height:1em, $weight:normal){
    font-family: 'Myriad Pro Regular', 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    font-size: $size;
    line-height: $height;
    font-weight: $weight;
}

@mixin baskerville($size:1em, $height:1em, $weight:normal){
   font-family: 'Baskerville Regular', 'Times New Roman', Times, serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}

@mixin athelas($size:1em, $height:1em, $weight:normal){
   font-family: 'Athelas Regular', Georgia, 'Times New Roman', Times, serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}

@mixin quatro($size:1em, $height:1em, $weight:normal){
   font-family: 'Quatro Regular' 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}

@mixin helvetica($size:1em, $height:1em, $weight:normal){
   font-family: Helvetica, Arial, sans-serif;
   font-size: $size;
   line-height: $height;
   font-weight: $weight;
}

@mixin padd-def($bot: 20px, $top: 20px, $left:0px, $right:0px){
   padding-top: $top;
   padding-bottom: $bot;
   padding-left: $left;
   padding-right: $right;
}



.gallery-part{
    min-height: 1450px;
    @include flex(flex-start, stretch);
    flex-direction: column;


.gallery-intro{
   height: 604px;
   padding: 20px;
   @include flex();
   flex-direction: column;
 h2, p {
    max-width: 1518px;
    margin-top: 20px;
    text-align: center;
 }
 h2{
   margin-bottom: 40px;
   text-transform: uppercase;
   @include athelas(2.3em, 2.3em);
 }
 p{
     @include athelas(1em, 1em);
 }
}

.gallery{
    width: 100%;
    height: 840px;
}

.gallery-screen{
   overflow: hidden;
   height: 90%;
   width: 100%;
   position: relative;
 .screen{
    &, .title {
       transition-property: all;
       transition-duration: 0.5s;
       transition-timing-function: ease;
    }
    position: absolute;
    height: 100%;
    top:0;
    left:0;
     @include flex(flex-start, stretch);
    .gallery-item{
       background-repeat: no-repeat;
       background-size: cover;
       background-position: center;
       height:100%;
       @include flex(flex-end, flex-end);
       .title{
          text-transform: uppercase;
          padding: 20px;
          position: relative;
          margin-bottom:50px;
          top:150px;
       }
    }
    .gallery-item.active{
       .title{
          top:0;
       }
    }
 }
}

.gallery-control{
   height: 10%;
   width: 100%;
   @include flex();
  .positioner{
     padding: 0 10px;
     height: 100%;
     @include flex(space-between, center);
  }
}

}



.gallery-part.default{
   .gallery-intro {
      h2{
         color: $orange;
      }
   }
    .gallery-control{
      background-color: $whiten-gray;
      .btn{
         &:before{
            border-color: $orange;
         }
         box-shadow: 0 0 0 2px $orange;
         &:hover{
            background-color: $orange;
            &:before{
               border-color: $white;
            }
         }
      }
   }
    .gallery-screen{
      background: rgb(254,254,254);
background: linear-gradient(90deg, rgba(254,254,254,1) 25%, rgba(238,238,238,1) 25%,
 rgba(181,181,186,1) 50%, rgba(151,151,154,1) 50%,
 rgba(142,142,145,1) 50%, rgba(60,60,62,1) 75%, rgba(45,44,44,1) 75%);

 
   }
}



.gallery-part.dark{
   background-color: $black;
   .gallery-intro {
      h2{
         color: $orange;
      }
      p{
         color:$white;
      }
   }
    .gallery-control{
      background-color: $black;
      .btn{
         &:before{
            border-color: $orange;
         }
         box-shadow: 0 0 0 2px $orange;
         &:hover{
            background-color: $orange;
            &:before{
               border-color: $white;
            }
         }
      }
   }
    
}


.gallery-part.light{
   .gallery-intro {
      h2{
         color: $orange;
      }
   }
    .gallery-control{
      box-shadow: inset 0 0 0 2px $swampy;
      .btn{
         &:before{
            border-color: $orange;
         }
         box-shadow: 0 0 0 2px $orange;
         &:hover{
            background-color: $orange;
            &:before{
               border-color: $white;
            }
         }
      }
   }
    
}





@media screen and (max-width: 900px){
   .gallery-part .gallery{
      height: 600px;
   }
}




@media screen and (max-width: 500px) and (max-height: 700px){
   .gallery-part {
      min-height: 800px;
      .gallery{
      height: 400px;
      .btn{width: 25px; height: 25px;}
      }
      .gallery-intro{
         height: 400px;
      }
      .gallery-intro {
         h2{
            margin-bottom: 20px;
         }
         p{
            margin-top: 0px;
         }
      }
   }
   

   }
  
   