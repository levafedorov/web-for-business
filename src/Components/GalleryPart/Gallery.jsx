import React, { Component } from 'react';
import GalleryItem from './GalleryItem';
import Button from '../Buttons/Button';
import Positioner from '../Page/Positioner';
import PropTypes from 'prop-types';

export default class Gallery extends Component{
   

  state = {
     position: 0
  }
 
 
 leftHandler = (ev) => {
   const position = this.state.position;
   const max = this.props.photos.length - 1;
   if(position){
       this.setState({position: position - 1});
   }else{
    this.setState({position: max});
   }
 }


 rightHandler = (ev) => {
  const position = this.state.position;
  const max = this.props.photos.length - 1;
  if(position !== max){
      this.setState({position: position + 1});
  }else{
    this.setState({position: 0});
  }
}

  
 render(){

   const {position} = this.state;
  

   const {photos} = this.props;
   const total = photos.length;
   const itemWidth = (100 / total).toFixed(2);
   const screen =  {width: `${total * 100}%`,
                    transform: `translateX(-${itemWidth * position}%)`};
   

   return <div className="gallery" id="gallery">
       <div className="gallery-screen">
      <div className="screen" style={screen}>
      {photos.map((item, i) => <GalleryItem {...item} width={itemWidth} active={i === position} key={'photo-' + i}/>)}
      </div>
       </div>
       <div className="gallery-control">
       <Positioner>
       <Button class="rounded square-left" fn={this.leftHandler}/>
       <Button class="rounded square-right" fn={this.rightHandler}/>
       </Positioner>
       </div>
   </div>
 }
}

Gallery.defaultProps = {
 photos: [{photo:'', title:""}]
}