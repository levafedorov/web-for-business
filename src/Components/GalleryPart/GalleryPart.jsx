import React, {Component} from 'react';
import GalleryIntro from './GalleryIntro';
import Gallery from './Gallery';
import PageThemes from '../PageContexts/PageThemes';
import './GalleryPart.scss';
import PropTypes, { array, arrayOf, objectOf } from 'prop-types';


export const photos = [{photo:'/photos/photo-1.jpg', title:"hello there"}, 
{photo:'/photos/photo-2.jpg', title:"hello there"},
{photo:'/photos/photo-3.jpg', title:"hello there"}];



export class GalleryPart extends Component{


   static contextType = PageThemes;
 

   render(){
   let className = "gallery-part stretched transition-all "
   className += this.context.theme;  
    
   

    return <div className={className} id="portfolio">
        <GalleryIntro />
        <Gallery photos={photos}/>
        </div>
   }
}

GalleryPart.contextTypes = {
    theme: PropTypes.string,
    change: PropTypes.func
}

// GalleryPart.propTypes = {
//    photos: arrayOf(objectOf(PropTypes.string)).isRequired
// }

 