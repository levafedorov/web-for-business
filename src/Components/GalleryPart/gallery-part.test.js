import React from 'react';
import {shallow} from 'enzyme';
import {GalleryPart} from './GalleryPart';




describe('Gallery component', () => {
   let component;
   let instance;
   const context = {theme: 'default',  change: () => {}}

   beforeEach(() => {
     component = shallow(<GalleryPart/>);
     instance =  component.instance();     
   });

    test('Check different context', () => {
        const component = shallow(<GalleryPart/>, {context});
        expect(component).toMatchSnapshot();
        component.setContext({theme: 'light',  change: () => {}});
        expect(component).toMatchSnapshot();  
    });

});