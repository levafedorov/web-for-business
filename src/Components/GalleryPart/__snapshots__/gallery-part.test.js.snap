// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component Check different context 1`] = `
<div
  className="gallery-part stretched transition-all default"
  id="portfolio"
>
  <GalleryIntro />
  <Gallery
    photos={
      Array [
        Object {
          "photo": "/photos/photo-1.jpg",
          "title": "hello there",
        },
        Object {
          "photo": "/photos/photo-2.jpg",
          "title": "hello there",
        },
        Object {
          "photo": "/photos/photo-3.jpg",
          "title": "hello there",
        },
      ]
    }
  />
</div>
`;

exports[`Gallery component Check different context 2`] = `
<div
  className="gallery-part stretched transition-all light"
  id="portfolio"
>
  <GalleryIntro />
  <Gallery
    photos={
      Array [
        Object {
          "photo": "/photos/photo-1.jpg",
          "title": "hello there",
        },
        Object {
          "photo": "/photos/photo-2.jpg",
          "title": "hello there",
        },
        Object {
          "photo": "/photos/photo-3.jpg",
          "title": "hello there",
        },
      ]
    }
  />
</div>
`;
